src/ProgressOnderwijsUtils/Extensions/EnumerableExtensions.cs

```diff
@@ -142,7 +142,13 @@ public static bool ContainsDuplicates<T>([NotNull] this IEnumerable<T> list)
         public static bool ContainsDuplicates<T>([NotNull] this IEnumerable<T> list, IEqualityComparer<T> comparer)
         {
             var set = new HashSet<T>(comparer);
-            return !list.All(set.Add);
+            foreach (var item in list) {
+                if (!set.Add(item)) {
+                    return true;
+                }
+            }
+
+            return false;

```

Comment:
Is dit echt sneller denk je?

Reply:
De oude code is misbruik van de All-method.

Reply:
ik zou de oude geen misbruik willen noemen - het is wat ongewoon om side-effects te doen, maar niet verboden (en dat soort richtlijnen op een hele lokale manier overschijden is denk ik niet een probleem).

Maar dit is idd sneller; en naar mijn idee zijn "complexiteiten" die heel lokaal zijn eigenlijk geen probleem.

Reply:
ik zou het wel misbruik willen noemen